contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,445 @@
+  ```
+
+  For function return types, default to always typing them unless a function is simple enough to reason about its return type.
+
+  > Why? Explicit return type helps catch errors when implementation of the function changes. It also makes it easy to read code even when TypeScript intellisense is not provided.
+
+  ```ts
+  function simpleFunction(name: string) {
+    return `hello, ${name}`;
+  }
+
+  function complicatedFunction(name: string): boolean {
+    // ... some complex logic here ...
+    return foo;
+  }
+  ```
+
+<a name="conventions-jsdoc"></a><a name="1.10"></a>
+
+- [1.10](#conventions-jsdoc) **JSDoc**: Omit comments that are redundant with TypeScript. Do not declare types in `@param` or `@return` blocks. Do not write `@implements`, `@enum`, `@private`, `@override`
+
+  ```ts
+  // bad
+  /**
+   * @param {number} age
+   * @returns {boolean} Whether the person is a legal drinking age or nots
+   */
+  function canDrink(age: number): boolean {
+    return age >= 21;
+  }
+
+  // good
+  /**
+   * @param age
+   * @returns Whether the person is a legal drinking age or nots
+   */
+  ```
+
+<a name="convension-proptypes-and-defaultprops"></a><a name="1.11"></a>
+
+- [1.11](#convension-proptypes-and-defaultprops) **`propTypes` and `defaultProps`**: Do not use them. Use object destructing to assign default values if necessary.
+
+  > Refer to [the propTypes Migration Table](./PROPTYPES_CONVERSION_TABLE.md) on how to type props based on existing `propTypes`.
+
+  ```tsx
+  type MyComponentProps = {
+    requiredProp: string;
+    optionalPropWithDefaultValue?: number;
+    optionalProp?: boolean;
+  };
+
+  function MyComponent({
+    requiredProp,
+    optionalPropWithDefaultValue = 42,
+    optionalProp,
+  }: MyComponentProps) {
+    // component's code
+  }
+  ```
+
+<a name="convension-utility-types"></a><a name="1.12"></a>
+
+- [1.12](#convension-utility-types) **Utility Types**: Use types from [TypeScript utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html) and [`type-fest`](https://github.com/sindresorhus/type-fest) when possible.
+
+  ```ts
+  type Foo = {
+    bar: string;
+  };
+
+  // good
+  type ReadOnlyFoo = Readonly<Foo>;
+
+  // bad
+  type ReadOnlyFoo = {
+    readonly [Property in keyof Foo]: Foo[Property];
+  };
+  ```
+
+<a name="convension-object-type"></a><a name="1.13"></a>
+
+- [1.13](#convension-object-type) **`object`**: Don't use `object` type.
+
+  > Why? `object` refers to "any non-primitive type," not "any object". Typing "any non-primitive value" is not commonly needed.
+
+  ```ts
+  // bad
+  const foo: object = [1, 2, 3]; // TypeScript does not error
+  ```
+
+  If you know that the type of data is an object but don't know what properties or values it has beforehand, use `Record<string, unknown>`.
+
+  > Even though `string` is specified as a key, `Record<string, unknown>` type can still accepts objects whose keys are numbers or symbols. This is because number and
```

Comment:
It's not true that symbols are allowed as keys with `Record<string, unknown>` - [source](https://github.com/Expensify/App/pull/21050#discussion_r1238415220). @hayata-suenaga 

From previous TS doc:

> Note that numeric keys are allowed with the `Record<string, unknown>` type. Numeric keys are implicitly converted to strings during property assignment and access. To read more about this, see [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/2/objects.html#dynamically-adding-properties).

Reply:
forgot to update that part thank you for pointing that out

Reply:
changed